app/services/data_decoder.py

```diff
@@ -163,29 +163,26 @@ async def get_contract_abi(
         return await Contract.get_abi_by_contract_address(HexBytes(address), chain_id)
 
     @alru_cache(maxsize=2048)
-    async def get_contract_fallback_function(
+    async def has_contract_fallback_function(
         self, address: Address, chain_id: int | None
-    ) -> ABIFunction | None:
+    ) -> bool:
         """
         :param address: Contract address
         :param chain_id: Chain for the contract
-        :return: Fallback `ABIFunction` if found, `None` otherwise.
+        :return: True if found Fallback `ABIFunction`, False otherwise.
             If contract is not found for the chain, return the first one that matches in other chain.
         """
         abi = await self.get_contract_abi(address, chain_id)
         if not abi and chain_id is not None:
             # Try to find an ABI in other network
             abi = await self.get_contract_abi(address, None)
         if abi:
-            return next(
-                (
-                    cast(ABIFunction, dict(fn_abi, name="fallback"))
-                    for fn_abi in abi
-                    if fn_abi.get("type") == "fallback"
-                ),
-                None,
+            return bool(
+                dict(fn_abi, name="fallback")
```

Comment:
I didn't realize this conversion was not needed

```suggestion
                fn_abi
```